File: 1/1.8/Python/1.8.py
import random

def removeNills(matrix):
    rowsToDelete = set()
    colsToDelete = set()

    for i in range(len(matrix)):
        for j in range(len(matrix[0])):
            if matrix[i][j] == 0:
                rowsToDelete.add(i)
                colsToDelete.add(j)
              
    
    while len(rowsToDelete) > 0:
    
        x = rowsToDelete.pop()
        matrix.pop(x)
        
        newRowsToDelete = set()
        for row in rowsToDelete:
            if row > x:
                newRowsToDelete.add(row - 1)
            else:
                newRowsToDelete.add(row)
        
        rowsToDelete = newRowsToDelete

    while len(colsToDelete) > 0:
    
        y = colsToDelete.pop()

        for i in range(len(matrix)):
            matrix[i].pop(y)
        
        newColsToDelete = set()

        for col in colsToDelete:
            if col > y:
                newColsToDelete.add(col - 1)
            else:
                newColsToDelete.add(col)
        
        colsToDelete = newColsToDelete
    

def  printMatrix(matrix) :
    for i in range(len(matrix)):
        for j in range(len(matrix[0])):
            print(str(matrix[i][j]) + " ", end="")
        
        print("\n", end="")
    print("\n", end="")


if __name__ == '__main__':    
 
    N = 5
    M = 10
    matrix = [[0 for _ in range(M)] for _ in range(N)]

    for i in range(N):
        for j in range(M):
            matrix[i][j] = random.randint(0,9)
        
    printMatrix(matrix)
    removeNills(matrix)
    printMatrix(matrix)

File: 1/1.8/Go/1.8.go
package main

import (
	"fmt"
	"maps"
	"math/rand"
	"strconv"
)

func removeNills(matrix *[][]int) {

	rowsToDelete := make(map[int]bool)
	colsToDelete := make(map[int]bool)

	for i := 0; i < len(*matrix); i++ {
		for j := 0; j < len((*matrix)[0]); j++ {
			if (*matrix)[i][j] == 0 {
				rowsToDelete[i] = true
				colsToDelete[j] = true
			}
		}
	}
	for len(rowsToDelete) > 0 {
		var x int
		for k := range maps.Keys(rowsToDelete) {
			x = k
			break
		}

		*matrix = append((*matrix)[:x], (*matrix)[x+1:]...)

		delete(rowsToDelete, x)

		newRowsToDelete := make(map[int]bool)
		for row, _ := range rowsToDelete {
			if row > x {
				newRowsToDelete[row-1] = true
			} else {
				newRowsToDelete[row] = true
			}
		}
		rowsToDelete = newRowsToDelete

	}

	for len(colsToDelete) > 0 {
		var y int
		for k := range maps.Keys(colsToDelete) {
			y = k
			break
		}

		for i := 0; i < len(*matrix); i++ {
			(*matrix)[i] = append((*matrix)[i][:y], (*matrix)[i][y+1:]...)
		}

		delete(colsToDelete, y)

		newColsToDelete := make(map[int]bool)

		for col, _ := range colsToDelete {
			if col > y {
				newColsToDelete[col-1] = true
			} else {
				newColsToDelete[col] = true
			}
		}
		colsToDelete = newColsToDelete
	}

}

func printMatrix(matrix [][]int) {

	for i := 0; i < len(matrix); i++ {
		for j := 0; j < len(matrix[0]); j++ {
			fmt.Print(strconv.Itoa(matrix[i][j]) + " ")
		}
		fmt.Print("\n")
	}
	fmt.Print("\n")
}

func main() {

	N := 5
	M := 10

	var matrix [][]int

	for i := 0; i < N; i++ {
		matrix = append(matrix, []int{})
		for j := 0; j < M; j++ {
			matrix[i] = append(matrix[i], rand.Intn(10))
		}
	}

	printMatrix(matrix)
	removeNills(&matrix)
	printMatrix(matrix)

}
